src/app/services/bucket/s3.py

```diff
@@ -24,12 +24,16 @@ class S3Bucket:
         access_key: the S3 access key
         secret_key: the S3 secret key
         bucket_name: the bucket name
+        use_proxy: whether to use a proxy for public url generation
+        proxy_url: the proxy url
     """
 
-    def __init__(self, region: str, endpoint_url: str, access_key: str, secret_key: str, bucket_name: str) -> None:
+    def __init__(self, region: str, endpoint_url: str, access_key: str, secret_key: str, bucket_name: str, use_proxy: bool, proxy_url: str) -> None:
         _session = boto3.Session(access_key, secret_key, region_name=region)
         self._s3 = _session.client("s3", endpoint_url=endpoint_url)
         self.bucket_name = bucket_name
+        self.use_proxy = use_proxy
+        self.proxy_url = proxy_url
```

Comment:
Those two should be optional if I get the purpose of this (or default to false for use_proxy, and optional for proxy_url)